src/order/types.ts

```diff
@@ -34,7 +34,7 @@ export abstract class Order {
    * Returns the data for generating the maker EIP-712 permit signature
    * @return The data for generating the maker EIP-712 permit signature
    */
-  abstract permitData(): PermitTransferFromData;
+  abstract permitData(): PermitTransferFromData | PermitBatchTransferFromData;
```

Comment:
any downsides of splitting this into two functions?

Reply:
permitData and permitBatchData ?

Reply:
Think carefully about this one, I think this may be disruptive to integrators also as union typings are not well supported from imported ts libraries

Reply:
one approach to improve this might be to have a descriminating field? so like
```
type PermitData = {
   permitDataType: PermitDataType.SINGLE,
   data: PermitTransferFromData,
} | {
  permitDataType: PermitDataType.BATCH,
  data: PermitTransferFromBatchData,
}
```

This way at least integrators can discrimiante using the type field